akka-stream-tests/src/test/scala/akka/stream/impl/fusing/ActorGraphInterpreterSpec.scala

```diff
@@ -70,6 +80,52 @@ class ActorGraphInterpreterSpec extends AkkaSpec {
 
     }
 
+    "be able to interpret and resuse a simple bidi stage" in assertAllStagesStopped {
+      val identityBidi = new GraphStage[BidiShape[Int, Int, Int, Int]] {
+        val in1 = Inlet[Int]("in1")
+        val in2 = Inlet[Int]("in2")
+        val out1 = Outlet[Int]("out1")
+        val out2 = Outlet[Int]("out2")
+        val shape = BidiShape(in1, out1, in2, out2)
+
+        override def createLogic: GraphStageLogic = new GraphStageLogic {
+          setHandler(in1, new InHandler {
+            override def onPush(): Unit = push(out1, grab(in1))
+
+            override def onUpstreamFinish(): Unit = complete(out1)
+          })
+
+          setHandler(in2, new InHandler {
+            override def onPush(): Unit = push(out2, grab(in2))
+
+            override def onUpstreamFinish(): Unit = complete(out2)
+          })
+
+          setHandler(out1, new OutHandler {
+            override def onPull(): Unit = pull(in1)
+
+            override def onDownstreamFinish(): Unit = cancel(in1)
+          })
+
+          setHandler(out2, new OutHandler {
+            override def onPull(): Unit = pull(in2)
+
+            override def onDownstreamFinish(): Unit = cancel(in2)
+          })
+        }
+
+        override def toString = "IdentityBidi"
+      }
+
+      val identityBidiF = BidiFlow.wrap(identityBidi)
+      val identity = (identityBidiF atop identityBidiF atop identityBidiF).join(Flow[Int].map { x ⇒ x })
+
+      Await.result(
+        Source(1 to 10).via(identity).grouped(100).runWith(Sink.head),
+        3.seconds) should ===(1 to 10)
```

Comment:
Hm that'd need implicits on Source/Flow, we can do that later I guess - for now it's a bit more utility methods on the TestSink, see here: https://github.com/akka/akka/pull/18503